modules/sagemaker/sagemaker-studio/README.md

```diff
@@ -0,0 +1,99 @@
+
+1. SageMaker Studio Domain requires, along with
+2. IAM roles which would be linked to SM Studio user profiles. User Profile creating process is managed by manifests files in `manifests/shared-infra/mlops-modules.yaml`. You can simply add new entries in the list to create a new user. The user will be linked to a role depending on which group you add them to (`data_science_users` or `lead_data_science_users`).
+
+```
+  - name: data_science_users
+    value:
+    - data-scientist
+  - name: lead_data_science_users
+    value:
+    - lead-data-scientist
+```
+
+3. Default SageMaker Project Templates are also enabled on the account on the targeted region using a custom resource; the custom resource uses a lambda function, `functions/sm_studio/enable_sm_projects`, to make necessary SDK calls to both Amazon Service Catalog and Amazon SageMaker.
+
+## Inputs and outputs:
+### Required inputs:
+  - `VPC_ID`
+  - `subnet_ids`
+### Optional Inputs:
+  - `studio_domain_name`
+  - `studio_bucket_name`
+### Outputs (module metadata):
+  - `StudioDomainName`
+  - `StudioBucketName`
+  - `DataScientistRoleArn`
+  - `DataScientistRoleSSMName`
+  - `LeadDataScientistRoleArn`
+  - `LeadDataScientistRoleSSMName`
+  - `SageMakerExecutionRoleSSMName`
+  - `SageMakerExecutionRoleArn`
+## Getting Started
+
+### Prerequisites
+
+This is an AWS CDK project written in Python 3.8. Here's what you need to have on your workstation before you can deploy this project. It is preferred to use a linux OS to be able to run all cli commands and avoid path issues.
+
+* [Node.js](https://nodejs.org/)
+* [Python3.8](https://www.python.org/downloads/release/python-380/) or [Miniconda](https://docs.conda.io/en/latest/miniconda.html)
+* [AWS CDK v2](https://aws.amazon.com/cdk/)
+* [AWS CLI](https://aws.amazon.com/cli/)
+* [Docker](https://docs.docker.com/desktop/)
+
+### Module Structure
+
+```
+.
+├── LICENSE.txt
+├── Makefile
+├── README.md
+├── app.py
+├── cdk.context.json
+├── cdk.json
+├── functions                                   <--- lambda functions and layers
+│   └── sm_studio                               <--- sagemaker studio stack related lambda function
+│       └── enable_sm_projects                  <--- lambda function to enable sagemaker projects on the account and links the IAM roles of the domain users (used as a custom resource)
+├── sagemaker_studio
+│   ├── constructs
+│   │   └── sm_roles.py                         <--- construct containing IAM roles for sagemaker studio users
+│   │   └── networking.py                         <--- construct for networking
+│   ├── cdk_helper_scripts
+│   └── sagemaker_studio_stack.py               <--- stack to create sagemaker studio domain along with related IAM roles and the domain users
+├── requirements-dev.txt
+├── requirements.txt                            <--- cdk packages used in the stacks (must be installed)
+```
+## Troubleshooting
+
+
+* **Resource being used by another resource**
```

Comment:
I do not understand this section (though I do understand the need to troubleshoot).